test/puppetlabs/puppetdb/http/reports_test.clj

```diff
@@ -124,6 +124,30 @@
                                ["=" "certname" (:certname basic)]])
        #{(select-keys basic [:hash :certname :transaction_uuid])}))))
 
+(deftestseq query-metrics
+  [[version endpoint] [[:v4 (partial format "/v4/reports/%s/metrics")]]]
+
+  (let [basic         (:basic reports)
+        report-hash   (:hash (store-example-report! basic (now)))
+        basic (assoc basic :hash report-hash)]
+
+    (testing "metrics endpoint returns the proper data"
+      (response-equal?
+       (get-response (endpoint report-hash) ["=" "hash" report-hash])
+       (set (:metrics basic))))))
+
+(deftestseq query-logs
+  [[version endpoint] [[:v4 (partial format "/v4/reports/%s/logs")]]]
+
+  (let [basic         (:basic reports)
+        report-hash   (:hash (store-example-report! basic (now)))
```

Comment:
wrapping the test definitions in a let with the reused stuff is another thing we do in places